pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -155,9 +156,30 @@ func DetermineVolumeAction(pod *v1.Pod, desiredStateOfWorld cache.DesiredStateOf
 		// should be detached or not
 		return keepTerminatedPodVolume
 	}
+	if isPodNeededToBeRemoved(pod) {
```

Comment:
the containers status checking is done in `IsPodTerminated`. which is also called in `DetermineVolumeAction`.
should we still add container status checking After grace period?

Reply:
Is there a reason where `DeletionTimestamp` will be set for pods that are not being terminated? again - I think we should verify with sig-node and confirm at what point, we should consider a pod as "deleted" and hence proceed with detaching volumes it was using. cc @sjenning @derekwaynecarr

Reply:
I think this may not be enough to fix the bug, because while this function will cause removal of the pod from DSWP, I think the next function that adds the pods will add it right back and volume will not get detached. Lets test this on a real cluster and see if it works.

Reply:
thanks for pointing this out @gnufied ,
and @verult mentioned that there is a meeting about this, let's hold it now and if this PR is the way we want to go, i will update it.